templates/new_dataset_script.py

```diff
@@ -153,24 +153,24 @@ def _split_generators(self, dl_manager):
             ),
         ]
 
-    def _generate_examples(self, filepath, split):
+    def _generate_examples(self, **gen_kwargs):
```

Comment:
I agree it makes the connection between what's defined in the SplitGenerator gen_kwargs clearer when reading the template. But when implementing an actual dataset it's better to not have kwargs.

Plus it's already mentioned in each SplitGenerator definition that
```
# These kwargs will be passed to _generate_examples
``` 
and also in `_generate_examples`:
```
# TODO: This method will receive as arguments the `gen_kwargs` defined in the previous `_split_generators` method.
```

Maybe we can add another comment:
```suggestion
    def _generate_examples(self, filepath, split):  # you can add arguments to this method by passing them in the `gen_kwargs`
```

Let me know what you think

Reply:
Yeah I'll definitely concede that it's already written in the clear - maybe it's more that I've not read python for a while rather than the template being ambiguous.

> But when implementing an actual dataset it's better to not have kwargs.

Yup I agree for sure. Weak opinion: all the examples in the repo have named parameters and set the tone, while the template is for onboarding and should favour clearer "magic" where possible. 
But all in all my aversion for kwargs prevents me from pushing too much in that direction - will push an additional comment